Python/optimizer.c

```diff
@@ -272,14 +271,23 @@ counter_optimize(
     int Py_UNUSED(curr_stackentries)
 )
 {
+    int oparg = instr->op.arg;
+    while (instr->op.code == EXTENDED_ARG) {
+        instr++;
+        oparg = (oparg << 8) | instr->op.arg;
+    }
+    if (instr->op.code != JUMP_BACKWARD) {
+        PyErr_Format(PyExc_ValueError, "Counter optimizer can only handle backward edges");
+        return -1;
+    }
     _PyCounterExecutorObject *executor = (_PyCounterExecutorObject *)_PyObject_New(&_PyCounterExecutor_Type);
     if (executor == NULL) {
         return -1;
     }
     executor->executor.execute = counter_execute;
     Py_INCREF(self);
     executor->optimizer = (_PyCounterOptimizerObject *)self;
-    executor->next_instr = instr;
+    executor->next_instr = instr + 2 - oparg;
```

Comment:
Makes it a little more future-proof (and self-documenting):
```suggestion
    executor->next_instr = instr + 1 + _PyOpcode_Caches[JUMP_BACKWARD] - oparg;
```